shared/ensemble_building/ensemble_selection.py

```diff
@@ -0,0 +1,305 @@
+import random
+from collections import Counter
+from typing import Callable, List, Tuple, Union
+
+import numpy as np
+
+from sklearn.pipeline import Pipeline
+
+from shared.ensemble_building.abstract_ensemble import AbstractEnsemble
+
+
+class EnsembleSelection(AbstractEnsemble):
+    """Implementation of Ensemble Selection from the paper:
+    Ensemble selection from libraries of models, Caruana 2016.
+
+    This class builds a set of weights that are used to aggregate
+    individual scikit-learn pipelines's prediction.
+
+    Attributes
+    ----------
+    ensemble_size : int
+        The maximum number of individual members that an ensemble can have.
+
+    calculate_loss: Callable
+        A function with signature calculate_loss(solutions, predictions) that estimate
+        the loss of the current predictions vs the expected solutions. The current
+        implementation of ensemble selections solves a minimization problem.
+    random_state: np.random.RandomState
+        A random number generator used to break ties when 2 models provide the same
+        performance gain.
+    bagging: bool
+        Enables the bagging approach from Ensemble selection from libraries
+        of models, Caruana 2016.
+    mode: str ['fast', 'slow']
+        Selects between the fast and slow (but intuitive) implementation of ensemble
+        selection
+
+    Returns
+    -------
+    self
+    """
+    def __init__(
+        self,
+        ensemble_size: int,
+        calculate_loss: Callable,
```

Comment:
Will this be updated once the metric module is copied, too?

Reply:
Maybe it would be good to also already initialize the metric module so all required dependencies are already there and this initial PR can be merged in one go?